website/project/views/contributor.py

```diff
@@ -867,21 +876,40 @@ def claim_user_form(auth, **kwargs):
 def _add_related_claimed_tag_to_user(pid, user):
     """
     Adds claimed tag to incoming users, depending on whether the resource has related source tags
-    :param pid: guid of either the node or the preprint
+    :param pid: guid of either the node or the preprint or the registration
     :param user: the claiming user
     """
     node = AbstractNode.load(pid)
     preprint = Preprint.load(pid)
+    registration = Registration.load(pid)
     osf_claimed_tag, created = Tag.all_tags.get_or_create(name=provider_claimed_tag('osf'), system=True)
-    if node:
+    if registration:
+        registration_provider_id = registration.provider._id
+        registration_claimed_tag, created = Tag.all_tags.get_or_create(name=provider_claimed_tag(registration_provider_id, 'registry'), system=True)
+        user.add_system_tag(registration_claimed_tag)
+    elif node:
         node_source_tags = node.all_tags.filter(name__icontains='source:', system=True)
-        if node_source_tags.exists():
-            for tag in node_source_tags:
-                claimed_tag, created = Tag.all_tags.get_or_create(name=NODE_SOURCE_TAG_CLAIMED_TAG_RELATION[tag.name],
-                                                                  system=True)
-                user.add_system_tag(claimed_tag)
+        if node.is_collected:
+            collection_provider_ids = [collection_submission.collection.provider._id for collection_submission in node.collecting_metadata_qs.all()]
+            for collection_provider_id in collection_provider_ids:
+                collection_claimed_tag, _ = Tag.all_tags.get_or_create(name=provider_claimed_tag(collection_provider_id, 'collections'), system=True)
+                user.add_system_tag(collection_claimed_tag)
+            if node_source_tags.filter(name=CampaignSourceTags.Osf4m.value).exists():
```

Comment:
Same notes as above about catching everything vs. just osf4m in the collections case.